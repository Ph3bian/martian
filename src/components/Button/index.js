import React from "react";
// import PropTypes from "prop-types";
import styles from "./button.module.scss";

const Button = ({ type, children, text, onClick, variant, ...rest }) => (
  <button
    type={type || "button"}
    onClick={onClick}
    className={[styles.Button, styles[variant]].join(" ")}
    {...rest}
  >
    {text || children}
  </button>
);

// Button.propTypes = {
//   variant: PropTypes.string,
//   text: PropTypes.string,
//   type: PropTypes.string.isRequired,
//   onClick: PropTypes.func,
// };

// Button.defaultProps = {
//   variant: "",
//   text: "Submit",
//   onClick: () => null,
// };

export default Button;
