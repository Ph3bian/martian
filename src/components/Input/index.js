import React from "react";
import PropTypes from "prop-types";
import styles from "./input.module.scss";

const Input = ({ label, error, ...rest }) => (
  <div className={styles.Input}>
    <label>{label}</label>
    <div className={styles.InputField}>
      <input {...rest} className={error && "error"} />
      {error && (
        <div>
          <span className={styles.InputError}> {error}</span>
        </div>
      )}
    </div>
  </div>
);

Input.propTypes = {
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  rest: PropTypes.shape({}),
};

Input.defaultProps = {
  error: "",
  rest: null,
};
export default Input;
