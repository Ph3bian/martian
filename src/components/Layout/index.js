import React, { useEffect } from "react";
import styles from "./layout.module.scss";
import { SunShineIcon, MoonIcon } from "assets/svg";
import { handleTheme } from "./functions";

const Layout = ({ children }) => {
  const [icon, setIcon]= React.useState("moon")
  useEffect(() => {
    if (localStorage.getItem("theme")) {
      document.getElementById("root").setAttribute(
        "data-theme",
        localStorage.getItem("theme")
      );
    }
  }, []);

  return (
    <div className={styles.Layout}>
      <div className={styles.LayoutHeader}>
        Martian
        <div onClick={()=>handleTheme(setIcon)}>
          {icon === "moon" ? (
            <SunShineIcon />
          ) : (
            <MoonIcon />
          )}
        </div>
      </div>
      <div className={styles.LayoutBody}>{children}</div>
    </div>
  );
};
export default Layout;
