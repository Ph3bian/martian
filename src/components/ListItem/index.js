import React from "react";
import { ChevronRightIcon } from "assets/svg";
import styles from "./listItem.module.scss";
const ListItem = ({ handleClick, data }) => {
  return (
    <div className={styles.List}>
      {data &&
        data.map((item, id) => (
          <div
            className={styles.ListItem}
            onClick={() => handleClick(item)}
            key={`${item.name}_${item.budget_spent - id}`}
          >
            <div className={styles.ListItemDescription}>
              <h2>{item.name}</h2>
              <p>{item.budget}</p>
              <span>{item.budget_spent}</span>
            </div>
            <div className={styles.ListItemArrow}>
              <ChevronRightIcon />
            </div>
          </div>
        ))}
    </div>
  );
};
export default ListItem;
