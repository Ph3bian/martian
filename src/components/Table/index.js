import React from "react";
import styles from "./table.module.scss";
const Table = ({ data = [], headers = [], handleClick }) => {
  const handleRowClick = (item) => {
    return handleClick && handleClick(item);
  };
  return (
    <table className={styles.Table}>
      <thead>
        <tr>
          {headers.map(({ name, title }) => (
            <th key={name}>{title}</th>
          ))}
        </tr>
      </thead>
      <tbody>
        {data.map((item) => (
          <tr onClick={()=>handleRowClick(item)}>
            {headers.map((header) => (
              <td>
                {header.formatter ? header.formatter(item) : item[header.name]}
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default Table;
