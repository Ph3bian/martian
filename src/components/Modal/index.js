import React, { useRef, useCallback, useEffect } from "react";
import styles from "./modal.module.scss";
import { CloseIcon } from "../../assets/svg";
import Button from "../Button";

const Modal = ({
  children,
  title,
  setShown,
  shown,
  hasFooter,
  hasHeader = true,
  handleSubmit,
  loading,
  formId
}) => {
  const node = useRef(null);

  const handleHide = useCallback(
    (event) => {
      if (event.key === "Escape") {
        setShown(!shown);
        return;
      }
    },
    [setShown, shown]
  );

  useEffect(() => {
    document.addEventListener("keydown", handleHide);
    return () => {
      document.removeEventListener("keydown", handleHide);
    };
  }, [handleHide]);

  return (
    <div
      tabIndex="-1"
      role="dialog"
      className={styles.Modal}
      aria-labelledby="modalTitle"
      aria-describedby="modalDesc"
      data-show={shown}
    >
      <div className={styles.ModalContent} ref={node}>
        {hasHeader && (
          <div className={styles.header}>
            <h3 id="modalTitle">{title}</h3>
            <CloseIcon
              aria-label="Close"
              role="button"
              onKeyDown={() => setShown(!shown)}
              onClick={() => setShown(!shown)}
            />
          </div>
        )}
        <div className={styles.body}>
          <div className={styles.bodyContent} id="modalDesc">
            {children}
          </div>
        </div>
        <div data-footer={hasFooter} className={styles.footer}>
          <Button
            type="button"
            variant="neutral"
            onKeyDown={() => setShown(!shown)}
            onClick={() => setShown(!shown)}
          >
            Close
          </Button>
          <Button
            type="submit"
            data-dismiss="modal"
            form={formId}
            variant="primary"
            disabled={loading}
            name="submit"
          >
            {loading ? "loading..." : "Submit"}
          </Button>
        </div>
      </div>
    </div>
  );
};

export default Modal;
