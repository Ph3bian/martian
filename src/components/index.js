import Button from "./Button";
import Input from "./Input";
import Table from "./Table";
import ListItem from "./ListItem";
import Layout from "./Layout";
import Modal from "./Modal";

export { Button, Input, Layout, Table, Modal, ListItem };
