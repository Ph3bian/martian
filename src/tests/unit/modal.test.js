import React from 'react';
import { axe, toHaveNoViolations } from 'jest-axe';
import { render, fireEvent } from '@testing-library/react';
import { Modal }from 'components';

describe('The <Modal/> component', () => {
  expect.extend(toHaveNoViolations);

  const defaultProps = {
    title: "Test Modal",
    setShown: jest.fn() ,
    shown:true,
    hasFooter:true,
    hasHeader: true,
    handleSubmit: jest.fn(),
    loading:false,
  };

  test('Should render button correctly', () => {
    const { asFragment } = render(<Modal {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });


  test('Should not fail any accessibility tests', async () => {
    const { container } = render(<Modal {...defaultProps} />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
