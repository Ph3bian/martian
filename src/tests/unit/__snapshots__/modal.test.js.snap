// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <Modal/> component Should render modal correctly 1`] = `
<DocumentFragment>
  <div
    aria-describedby="modalDesc"
    aria-labelledby="modalTitle"
    class="Modal"
    data-show="true"
    role="dialog"
    tabindex="-1"
  >
    <div
      class="ModalContent"
    >
      <div
        class="header"
      >
        <h3
          id="modalTitle"
        >
          Test Modal
        </h3>
        <svg
          aria-label="Close"
          role="button"
        >
          close-icon.svg
        </svg>
      </div>
      <div
        class="body"
      >
        <div
          class="bodyContent"
          id="modalDesc"
        />
      </div>
      <div
        class="footer"
        data-footer="true"
      >
        <button
          class="Button neutral"
          type="button"
        >
          Close
        </button>
        <button
          class="Button primary"
          data-dismiss="modal"
          name="submit"
          type="button"
        >
          Submit
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
